src/modules/CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ set(MODULES_INSTALL_DIR "/usr/lib/osconfig" CACHE FILEPATH "Directory used for i
 add_subdirectory(commandrunner)
 add_subdirectory(firewall)
 add_subdirectory(networking)
+add_subdirectory(packagemanagerconfiguration)

```

Comment:
>packagemanagerconfiguration [](http://example.com/codeflow?start=17&length=27)

This name is excessively long and this results in the coed and file names for this module to be also super long. Can we replace it with a shorter name? 

How about: "PmConfig"? Let's discuss

Reply:
We agreed on that name in the MIM json file, so the code and file names are consistent with it.

Would you like to completely change this name also in the MIM json file? Or only file names and classes name?

Reply:
That long name is a description. Imagine if instead of OSConfig it would be OperatingSystemConfiguration.

Please change to a shorter name in the code and also in the MIM if you want them to match (they do not have to match exactly)

Reply:
OS is way more obvious that it means OperatingSystem vs PM meaning PackageManager.

PmConfig is not clear at all that it is referring to packages. Previously we had several other proposals:
- AptInstall
- PackageTool
- PackageUpdate
- PackageController
- PackageManager
- PackageHandler
- Packages
Could you let us know which one you would find more suitable?

/cc @MatthewMWR

Reply:
There are multiple interleaved considerations here, let's try separating them out first and then consolidating.

1. "Config" vs "Configuration": Subjectively I prefer Config (shorter, still obvious meaning, aligns with OSConfig name), but I thought at one point someone had asserted that abbreviations weren't allowed in the in object model? Am I remembering wrong? Is there such a rule? If so, should we consider changing it?

2. "PackageManager" vs "PM" vs "PacMan" vs "Apps" vs "Applications", vs "Components", etc. I think we had a long debate about this a month ago (approx), and settled on PackageManager for various reasons. For example, in OS/Linux context "apps" or "applications" have different connotations from "packages" which are often libraries. "Apt" is too specific some something that could be a cross-OS API/model, etc.

As of now, I feel most aligned to "PackageManagerConfig" as the name in the object model. I have no opinion about the names of classes, etc.

Reply:
The MIM name can remain "PackageManagerConfiguration". The MIM was already approved in a separate PR.

This is about the names used for the source code and the binaries.

I am proposing to do here similar to ZTSI. We also have a very long name here. Let's use PMC. Aka for example: "pmclib" etc.

Reply:
changed as suggested